client/auth/oauth-utils.js

```diff
@@ -0,0 +1,61 @@
+/**
+ * Internal dependencies
+ */
+import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
+import { bumpStat } from 'calypso/lib/analytics/mc';
+
+export const actions = {
+	AUTH_LOGIN: 'AUTH_LOGIN',
+	RECEIVE_AUTH_LOGIN: 'RECEIVE_AUTH_LOGIN',
+	AUTH_RESET: 'AUTH_RESET',
+};
+
+export const errors = {
+	ERROR_REQUIRES_2FA: 'needs_2fa', // From WP.com API
+	ERROR_INVALID_OTP: 'invalid_otp', // From WP.com API
+};
+
+export async function makeAuthRequest( username, password, authCode = '' ) {
+	const response = await globalThis.fetch( '/oauth', {
+		method: 'POST',
+		headers: { 'Content-Type': 'application/json' },
+		body: JSON.stringify( {
+			username,
+			password,
+			auth_code: authCode.replace( /\s/g, '' ),
```

Comment:
It doesn't really matter much, but I'd personally do the `.replace( /\s/g, '' )` call in the React UI code. Not in code that makes the REST request. Sanitizing the input value is a UI concern.

Reply:
Would 5d9e19f work for you?